Frends.HTTP.Request/Frends.HTTP.Request/Request.cs

```diff
@@ -0,0 +1,400 @@
+        var isContentAllowed = Definitions.Method.TryParse(method, ignoreCase: true, result: out SendMethod _);
+
+        using (var request = new HttpRequestMessage(new HttpMethod(method), new Uri(url))
+        {
+            Content = isContentAllowed ? content : null,
+        })
+        {
+
+            //Clear default headers
+            content.Headers.Clear();
+            foreach (var header in headers)
+            {
+                var requestHeaderAddedSuccessfully = request.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                if (!requestHeaderAddedSuccessfully && request.Content != null)
+                {
+                    //Could not add to request headers try to add to content headers
+                    // this check is probably not needed anymore as the new HttpClient does not seem fail on malformed headers
+                    var contentHeaderAddedSuccessfully = content.Headers.TryAddWithoutValidation(header.Key, header.Value);
+                    if (!contentHeaderAddedSuccessfully)
+                    {
+                        Trace.TraceWarning($"Could not add header {header.Key}:{header.Value}");
+                    }
+                }
+            }
+
+            HttpResponseMessage response;
+            try
+            {
+                response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
+            }
+            catch (TaskCanceledException canceledException)
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    // Cancellation is from outside -> Just throw 
+                    throw;
+                }
+
+                // Cancellation is from inside of the request, mostly likely a timeout
+                throw new Exception("HttpRequest was canceled, most likely due to a timeout.", canceledException);
+            }
+
+
+            // this check is probably not needed anymore as the new HttpClient does not fail on invalid charsets
+            if (options.AllowInvalidResponseContentTypeCharSet && response.Content.Headers?.ContentType != null)
+            {
+                response.Content.Headers.ContentType.CharSet = null;
+            }
+
+            return response;
+        }
+    }
+
+    #region Test methods
+    // For tests
+    internal static readonly ObjectCache ClientCache = MemoryCache.Default;
+
+    private static readonly CacheItemPolicy _cachePolicy = new CacheItemPolicy() { SlidingExpiration = TimeSpan.FromHours(1) };
+
+    internal static void ClearClientCache()
+    {
+        var cacheKeys = ClientCache.Select(kvp => kvp.Key).ToList();
+        foreach (var cacheKey in cacheKeys)
+        {
+            ClientCache.Remove(cacheKey);
+        }
+    }
+
+    internal static IHttpClientFactory ClientFactory = new HttpClientFactory();
+    #endregion
+}
+
+#region Helper methods
```

Comment:
I think all of the below methods can be moved to the Task class as helper methods.

Reply:
Created own classes for helper methods. Couldn't get just methods to work properly with tests.